routes/v1.js

```diff
@@ -502,52 +521,60 @@ api.declare({
     if (!_.isEqual(taskDef, def) || task.runs.length === 0) {
-      return res.reportError(
-        "RequestConflict", [
-          "taskId {{taskId}} already used by another task.",
-          "This could be the result of faulty idempotency!"
-        ].join('\n'),
-        {
-          taskId,
-        });
+      return res.reportError('RequestConflict', [
+        "taskId {{taskId}} already used by another task.",
+        "This could be the result of faulty idempotency!",
+        "Existing task definition was: {{existingTask}}",
+        "This request tried to define: {{taskDefinition}}",
+      ].join('\n'), {
+        taskId,
+        existingTask: def,
+        taskDefinition: taskDef,
+      });
     }
   }
 
-  // Construct task status, as we'll return this many times
-  var status = task.status();
+  // Track dependencies, if not already scheduled
+  if (task.state() === 'unscheduled') {
+    // Track dependencies, adds a pending run if ready to run
+    let err = await this.dependencyTracker.trackDependencies(task);
+    // We get an error here the task will be left in state = 'unscheduled',
+    // any attempt to use the same taskId will fail. And eventually the task
+    // will be resolved deadline-expired. But since createTask never returned
+    // successfully...
+    if (err) {
+      return res.reply('InputError', err.message, err.details);
+    }
+  }
 
-  // If first run isn't pending, all message must have been published before,
-  // this can happen if we came from the catch-branch (it's unlikely to happen)
-  if (task.runs[0].state !== 'pending') {
-    return res.reply({
-      status:   status
-    });
+  // Construct task status, as we'll return this many times
+  let status = task.status();
+
+  // If first run isn't unscheduled or pending, all message must have been
+  // published before, this can happen if we came from the catch-branch
+  // (it's unlikely to happen). But no need to publish messages again
+  let runZeroState = task.runs[0].state;
+  if (runZeroState !== 'unscheduled' && runZeroState !== 'pending') {
+    return res.reply({status});
   }
 
-  await Promise.all([
-    // Put message into the task pending queue
-    this.queueService.putPendingMessage(task, 0),
+  // Publish task-defined message, we want this arriving before the
+  // task-pending message, so we have to await publication here
+  await this.publisher.taskDefined({status}, task.routes);
 
-    // Publish pulse messages
-    (async () => {
-      // Publish task-defined message, we want this arriving before the
-      // task-pending message, so we have to await publication here
-      await this.publisher.taskDefined({
-        status:         status
-      }, task.routes);
+  // If first run is pending we publish messages about this
+  if (runZeroState === 'pending') {
+    await Promise.all([
+      // Put message into the task pending queue
+      this.queueService.putPendingMessage(task, 0),
```

Comment:
Just reading this as a novice to this codebase, what's the difference between this line and the subsequent one ([line 572](https://github.com/taskcluster/taskcluster-queue/pull/81/files#diff-ebe0840faded7b2e7bc8c6e798d1b159R572) at the moment).